aviatrix/resource_aviatrix_controller_private_mode_config.go

```diff
@@ -0,0 +1,191 @@
+package aviatrix
+
+import (
+	"context"
+	"strings"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v2/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAviatrixControllerPrivateModeConfig() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceAviatrixControllerPrivateModeConfigCreate,
+		ReadWithoutTimeout:   resourceAviatrixControllerPrivateModeConfigRead,
+		UpdateWithoutTimeout: resourceAviatrixControllerPrivateModeConfigUpdate,
+		DeleteWithoutTimeout: resourceAviatrixControllerPrivateModeConfigDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"enable_private_mode": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether to enable Private Mode on the Controller.",
+			},
+			"copilot_instance_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Copilot instance ID to associate with the Controller for Private Mode.",
+			},
+			"proxies": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "List of proxies.",
+			},
+		},
+	}
+}
+
+func resourceAviatrixControllerPrivateModeConfigCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	enablePrivateMode := d.Get("enable_private_mode").(bool)
+	if !enablePrivateMode {
+		for _, key := range []string{"copilot_instance_id", "proxies"} {
+			if _, ok := d.GetOk(key); ok {
+				return diag.Errorf("failed to create Controller Private Mode config: %q must be empty when %q is false", key, "enable_private_mode")
+			}
```

Comment:
That is an interesting point. One of the interesting features of using the Context methods and returning diag.Diagnostics is that we can return multiple errors at the same time. This could be something to consider here.

Another feature is that we can specify the exact attribute that caused the error, which should make fixing the error easier. I was testing something like this on microseg policies because if multiple policies have errors, it's hard to quickly fix it.